libs/layout/src/lib/services/media-query/mediaquery.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { WindowService } from '@ngx/utils';
+import { filter, map, Observable, ReplaySubject } from 'rxjs';
+
+/**
+ * A service that can be used to track media queries and their changes. It exposes a method
+ * to register media queries, which takes an array of tuples with the id of the media query
+ * and the query itself. The service will then emit the id of the media query that has
+ * changed when subscribed to the `getMatchingQuery$` method.
+ */
+@Injectable()
```

Comment:
@IbenTesara remarked:
> If we'd be supporting multiple matchingQueries, perhaps it would be more sensible to define the queries (and listeners) globally (in the `app.component.ts`), instead of setting them in each component that requires their use.